node_test.go

```diff
@@ -12,15 +12,25 @@ import (
 	ipld "github.com/ipfs/go-ipld-format"
 )
 
+var sampleCid cid.Cid
+
+func init() {
+	var err error
+	sampleCid, err = cid.Cast([]byte{1, 85, 0, 5, 0, 1, 2, 3, 4})
+	if err != nil {
+		panic(err)
+	}
+}
+
 func TestStableCID(t *testing.T) {
 	nd := &ProtoNode{}
 	nd.SetData([]byte("foobar"))
 	nd.SetLinks([]*ipld.Link{
-		{Name: "a"},
-		{Name: "b"},
-		{Name: "c"},
+		{Name: "a", Cid: sampleCid},
+		{Name: "b", Cid: sampleCid},
+		{Name: "c", Cid: sampleCid},
 	})
-	expected, err := cid.Decode("QmSN3WED2xPLbYvBbfvew2ZLtui8EbFYYcbfkpKH5jwG9C")
+	expected, err := cid.Decode("QmciCHWD9Q47VPX6naY3XsPZGnqVqbedAniGCcaHjBaCri")
```

Comment:
was the prior test valid DagPB? If so, are we get the same CID output as a way of making sure we haven't broken things?

What's the idea behind the change here? Is it just that we're testing more here? Same for the other modifications in this file.

Reply:
My understanding that go-codec-dagpb does not accept Links with empty CIDS. This is inline with the DagPB spec, which states:
"Hash:
- even though Hash is optional in the Protobuf encoding, it should not be treated as optional when creating new blocks or decoding existing ones, an omitted Hash should be interpreted as a bad block"

I suspect the old library accepted empty CIDs because it was written earlier. I agree though this may be of concern -- at least for compatibility checking, even if the old implementation is TOO flexible for the spec.

Reply:
Seems like a reasonable breakage to me, I'd certainly hope this couldn't matter in go-ipfs given that it's sort of non-sensical and spec-breaking.